fedopt_guide/cifar10/federated_trainer.py

```diff
@@ -0,0 +1,183 @@
+
+
+def _get_hparam_flags():
+  """Returns an ordered dictionary of pertinent hyperparameter flags."""
+  hparam_dict = utils_impl.lookup_flag_values(shared_flags)
+
+  # Update with optimizer flags corresponding to the chosen optimizers.
+  opt_flag_dict = utils_impl.lookup_flag_values(optimizer_flags)
+  opt_flag_dict = optimizer_utils.remove_unused_flags('client', opt_flag_dict)
+  opt_flag_dict = optimizer_utils.remove_unused_flags('server', opt_flag_dict)
+  hparam_dict.update(opt_flag_dict)
+
+  # Update with task-specific flags.
+  task_name = FLAGS.task
+  if task_name in TASK_FLAGS:
+    task_hparam_dict = utils_impl.lookup_flag_values(TASK_FLAGS[task_name])
+    hparam_dict.update(task_hparam_dict)
+
+  return hparam_dict
+
+
+def _get_task_args():
+  """Returns an ordered dictionary of task-specific arguments.
+  This method returns a dict of (arg_name, arg_value) pairs, where the
+  arg_name has had the task name removed as a prefix (if it exists), as well
+  as any leading `-` or `_` characters.
+  Returns:
+    An ordered dictionary of (arg_name, arg_value) pairs.
+  """
+  task_name = FLAGS.task
+  task_args = collections.OrderedDict()
+
+  if task_name in TASK_FLAGS:
+    task_flag_list = TASK_FLAGS[task_name]
+    task_flag_dict = utils_impl.lookup_flag_values(task_flag_list)
+    task_flag_prefix = TASK_FLAG_PREFIXES[task_name]
+    for (key, value) in task_flag_dict.items():
+      if key.startswith(task_flag_prefix):
+        key = key[len(task_flag_prefix):].lstrip('_-')
+      task_args[key] = value
+  return task_args
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError('Expected no command-line arguments, '
+                         'got: {}'.format(argv))
+
+  client_optimizer_fn = optimizer_utils.create_optimizer_fn_from_flags('client')
+  server_optimizer_fn = optimizer_utils.create_optimizer_fn_from_flags('server')
+
+  client_lr_schedule = optimizer_utils.create_lr_schedule_from_flags('client')
+  server_lr_schedule = optimizer_utils.create_lr_schedule_from_flags('server')
+
+  def iterative_process_builder(
+      model_fn: Callable[[], tff.learning.Model],
+      client_weight_fn: Optional[Callable[[Any], tf.Tensor]] = None,
+  ) -> tff.templates.IterativeProcess:
+    """Creates an iterative process using a given TFF `model_fn`.
+    Args:
+      model_fn: A no-arg function returning a `tff.learning.Model`.
+      client_weight_fn: Optional function that takes the output of
+        `model.report_local_outputs` and returns a tensor providing the weight
+        in the federated average of model deltas. If not provided, the default
+        is the total number of examples processed on device.
+    Returns:
+      A `tff.templates.IterativeProcess`.
+    """
+
+    return tff.learning.build_federated_averaging_process(
+        model_fn=model_fn,
+        client_optimizer_fn=client_optimizer_fn,
+        server_optimizer_fn=server_optimizer_fn,
+        client_weight_fn=client_weight_fn, 
```

Comment:
I believe this has changed to client_weighting in tff-nightly

Reply:
Just a general question, should I always refer to the nightly version?